addons/ttf/ttf.c

```diff
@@ -1005,10 +1005,10 @@ static bool ttf_get_glyph_dimensions(ALLEGRO_FONT const *f,
       }
    }
    cache_glyph(data, face, ft_index, glyph, false);
-   *bbx = glyph->offset_x;
+   *bbx = glyph->offset_x + 1;
    *bbw = glyph->region.w - 2;
```

Comment:
The text rendering shouldn't be affected since this function is not used for rendering (as far as I can see).

The reason for the change in the other numbers is that glyph->region adds a 1 pixel margin to the glyphs, so to get the exact glyph bounding box you need to add 1 to the offsets and substract 2 to the width and height.

Apparently someone remembered this fact but only substracted 2 from region.w (at least that's the only explanation I can think of...).

The old version gets bbh, bbx and bby off by 1 or 2 pixels from the real position of the bounding box for the glyph, and the proposed patch should give a tight bounding box (in my tests it worked nicely).